projects/rocm-core/rocm_getpath.cpp

```diff
@@ -22,24 +22,34 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 //
 ////////////////////////////////////////////////////////////////////////////////
+#include <cstring> /* String Operations */
+#include <cstdlib> /* Dynamic Memory Mgmt */
+#include <cstdio>  /* FILENAME_MAX */
+#if !defined(_WIN32) && !defined(_WIN64)
+  #include <climits> /* PATH_MAX */
+  #include <link.h>  /* ELF Dynamic Linking DS */
+  #include <dlfcn.h> /* Dynamic linker Operations */
+  #define RC_PATH_MAX PATH_MAX
+#else
+  #if defined(_WIN32) || defined(_WIN64)
+    #include <windows.h> /* MAX_PATH */
+    #define PATH_MAX MAX_PATH
+  #elif !defined(PATH_MAX)
+    #define PATH_MAX FILENAME_MAX
+  #endif
+  #define RC_PATH_MAX (PATH_MAX > 1024 && FILENAME_MAX > 1024 ? 1024 : (PATH_MAX < FILENAME_MAX ? PATH_MAX : FILENAME_MAX))
+#endif
 
-#include <string.h>
-#include <stdlib.h>
-#include <limits.h> /* PATH_MAX */
-#include <stdio.h>
-#include <link.h>
-#include <dlfcn.h>
 #include "rocm_getpath.h"
 
 /* Macro for NULL CHECK */
 #define NULL_CHECK(ptr) if(!ptr) return PathIncorrecPararmeters;
 
-
 /* Target Library Install Dir */
 #define TARGET_LIB_INSTALL_DIR TARGET_LIBRARY_INSTALL_DIR
 
 /* Target Library Name Buf Size */
-#define LIBRARY_FILENAME_BUFSZ (PATH_MAX+1)
+#define LIBRARY_FILENAME_BUFSZ RC_PATH_MAX+1
```

Comment:
Do we really need LIBRARY_FILENAME_BUFSZ variable. Cant we use RC_PATH_MAX variable itself